doc/usage/extensions/autodoc.rst

```diff
@@ -507,6 +507,15 @@ autodoc provides the following additional events:
       succeed and signature wasn't specified in the directive.
    :param return_annotation: function return annotation as a string of the form
       ``" -> annotation"``, or ``None`` if there is no return annotation
+   :param documenter: The :class:`sphinx.ext.autodoc.Documenter` instance.  See
+      `Removing Type Hints`_ example.
+
+   .. versionadded:: 2.1
```

Comment:
I couldn't put this under `:param documenter:`, it seems directives cannot be used there.  If we do go with this, I think it's important to draw explicit attention to the 8th parameter not being available prior to 2.1.  Because it's really easy to end up on sphinx docs for a newer version than you are using, and if we don't make this abundantly obvious people trying to use `autodoc-process-signature` will be very confused!